examples/fp8_attention.py

```diff
@@ -90,8 +97,19 @@ def fp8_attention_kernel(
                 p_fp8 = p.to(v.dtype)  # Convert to same FP8 type as V
 
                 # Accumulate attention @ V with FP8 GEMM
-                v_t = v_tile.transpose(0, 1)  # [tile_n, dim]
-                pv = torch.matmul(p_fp8, v_t).to(torch.float32)  # [tile_m, dim]
+                # torch._scaled_mm requires second matrix to be column-major
+                # v_tile is [dim, tile_n], we need [tile_n, dim] in column-major
+                v_t = v_tile.contiguous().t()  # [tile_n, dim] in column-major format
```

Comment:
This error arises from the lowering in PT right, this is no longer true on b200 for certain inputs and we want to remove the restrciton

Reply:
Would passing in a column-major matrix for operand B still work, for `torch._scaled_mm` on B200? It would be great to not need to change the Helion user code for different hardware.

Reply:
I see sounds good - maybe we can simplify this later with a higher-level `hl.matmul` API that encapsulates the differences, or just follow PyTorch API when it lands

Reply:
The layouts are specified by the MMA instruction so can't relax the restriction unfortunately (at least on hopper)

Reply:
Yes this is only for reference mode. Should we maybe have a `hl._scaled_mm` to relax the requirement and hide away the format transform?

Reply:
I imagine that if you don't have have the second input in col major for tl.dot you will just hit a layout transformation and it will ultimately be slower; https://github.com/pytorch/pytorch/issues/147336#issuecomment-2860534299

Reply:
Maybe we can hide the layout transformation within the `hl._scaled_mm` implementation, and generate the transpose before tl.dot

Reply:
Ohh I think triton will do all of this for, you its just that globally transposing before calling the kernel is faster than doing it on the fly